pkgs/development/python-modules/playwright/update.sh

```diff
@@ -30,18 +30,38 @@ replace_sha() {
 }
 
 prefetch_browser() {
-    nix store prefetch-file --json --hash-type sha256 --unpack "$1" | jq -r .hash
+  # nix-prefetch is used to obtain sha with `stripRoot = false`
+  # doesn't work on macOS https://github.com/msteen/nix-prefetch/issues/53
+  nix-prefetch -q "{ stdenv, fetchzip }: stdenv.mkDerivation rec { name=\"browser\"; src = fetchzip { url = \"$1\"; stripRoot = $2; }; }"
 }
 
 update_browser() {
     name="$1"
-    suffix="$2"
-    arm64_suffix="${3:-$2-arm64}"
+    platform="$2"
+    stripRoot="false"
+    if [ "$platform" = "darwin" ]; then
```

Comment:
I don't see any reason to do this in this pr, it won't affect anything at the moment

Reply:
> Otherwise we might regress on older darwin releases compared to the current

Can you elaborate on this, I don't quite understand how can we support old macos versions? 
Maybe we can leave existing browsers-mac for such setups, but it doesn't work on intel cpu anyway and on aarch it's browser version for macos on CI server I guess.

Reply:
> Can you elaborate on this, I don't quite understand how can we support old macos versions?

Sure, could have phrased this better! My line of thinking was: The current, impure `playwright install` probably handles `revisionOverrides` correctly, i.e. downloads a different binary on older macos releases.
If we don't handle this here, we always default to the binary for the latest macos supported in browser.json, no?

Reply:
> The current, impure playwright install probably handles revisionOverrides correctly, i.e. downloads a different binary on older macos releases.

Only if nixpkgs binary cache is not used, otherwise browser build for os from nix build machine will be downloaded. 

For now revision overrides really used only for webkit on eol macos versions 
```
    "webkit": {
      "revision": "2083",
      "revisionOverrides": {
        "mac10.14": "1446",
        "mac10.15": "1616",
        "mac11": "1816",
        "mac11-arm64": "1816",
        "mac12": "2009",
        "mac12-arm64": "2009"
      },
      "browserVersion": "18.0"
    },
```

In this pr browser build for macos-14 is used that's why I said that implementing support in the pr won't make any difference. For now I don't even know how to make builds for different macos versions in nix, so supporting overrides for something that can't be build in nixpkgs doesn't make much sense to me.

If you think we should make this deriviation depend on host machine macos version (and know how to implement this) then please let me know, I may consider adding support for this along with revisionOverrides.